app/src/main/AndroidManifest.xml

```diff
@@ -52,6 +54,12 @@
         </activity>
 
         <activity
+            android:name=".SignInActivity"
+            android:theme="@style/AppTheme"
+            />
```

Comment:
Thanks. Fixed it. About that buggy scenario - could I simply flush backstack with Intent flags? I think there is no point to let user go back to SignInActivity